File: skylotoneoneone_ui/src/services/SessionService.js
import http from "../http-common"; 

class SessionService {
  getAllSessions(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/session/sessions`, searchDTO);
  }

  get(sessionId) {
    return this.getRequest(`/session/${sessionId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/session?field=${matchData}`, null);
  }

  addSession(data) {
    return http.post("/session/addSession", data);
  }

  update(data) {
  	return http.post("/session/updateSession", data);
  }
  
  uploadImage(data,sessionId) {
  	return http.postForm("/session/uploadImage/"+sessionId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new SessionService();

File: src/main/java/com/skylotoneoneone/service/SessionService.java
package com.skylotoneoneone.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skylotoneoneone.domain.Session;
import com.skylotoneoneone.dto.SessionDTO;
import com.skylotoneoneone.dto.SessionSearchDTO;
import com.skylotoneoneone.dto.SessionPageDTO;
import com.skylotoneoneone.dto.SessionConvertCriteriaDTO;
import com.skylotoneoneone.service.GenericService;
import com.skylotoneoneone.dto.common.RequestDTO;
import com.skylotoneoneone.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface SessionService extends GenericService<Session, Integer> {

	List<Session> findAll();

	ResultDTO addSession(SessionDTO sessionDTO, RequestDTO requestDTO);

	ResultDTO updateSession(SessionDTO sessionDTO, RequestDTO requestDTO);

    Page<Session> getAllSessions(Pageable pageable);

    Page<Session> getAllSessions(Specification<Session> spec, Pageable pageable);

	ResponseEntity<SessionPageDTO> getSessions(SessionSearchDTO sessionSearchDTO);
	
	List<SessionDTO> convertSessionsToSessionDTOs(List<Session> sessions, SessionConvertCriteriaDTO convertCriteria);

	SessionDTO getSessionDTOById(Integer sessionId);







}





